### ISSUE TITLE
np.linalg.norm

### ISSUE THREAD

**Issue by henon** [SciSharp/NumSharp#239]
I am going to port np.linalg.norm(...)

**Comment by Oceania2018** [comment]
It's not same as ?
![image](https://user-images.githubusercontent.com/1705364/55639218-77e54f00-578e-11e9-8320-89364ddc6805.png)

**Comment by henon** [comment]
ok, you are right. that implements the L2 norm I need. I'll add the shortcut np.linalg.norm(...) just for Python  compatibility

**Comment by henon** [comment]
no, after checking the code against the numpy docs, actually, the 2-norm is not the same what has been implemented in normalize(). There is no squaring of coefficients. Is that a bug?

https://het.as.utexas.edu/HET/Software/Numpy/reference/generated/numpy.linalg.norm.html

**Comment by Oceania2018** [comment]
Yes, they're different.

**Comment by 8** [comment]
Hi,

I also noticed that the function is missing in `0.20.5`.
I think one could add a shortcut from `np.linalg.norm(a)` => `np.sqrt(a.dot(a))` instead, which should be equivalent.
Not sure about the overloads though.

Does this make sense to you?

Take care,
Martin

**Comment by Nucs** [comment]
@henon @8, following up on the old `normalize()`-vs-L2 discussion here: the real thing shipped in **0.70.0** (#628). `np.linalg.norm` now covers every `ord`/`axis`/`keepdims`, not just the L2 shortcut:

```csharp
// L2 norm of a vector
Console.WriteLine(np.linalg.norm(np.array(new[] { 3.0, 4 })));   // 5.0

// Frobenius norm of a matrix (the default ord)
Console.WriteLine(np.linalg.norm(np.eye(3)));                     // 1.7320508075688772
```

Vector norms and the Frobenius matrix norm compute in pure-managed Core; the spectral/nuclear orders (`2`, `-2`, `'nuc'`) use the OpenBLAS SVD. Closing.
